dataquality/integrations/seq2seq/auto.py

```diff
@@ -0,0 +1,263 @@
+
+    Example:
+        >>> sample = {
+        ...     "instruction": "Summarize the following paragraph",
+        ...     "input": "The quick brown fox jumped over the lazy dog.",
+        ...     "target": "The quick brown fox jumped over the lazy dog.",
+        ... }
+        >>> format_alpaca(sample)
+        {
+            "formatted_input": (
+                "Human: Summarize the following paragraph "
+                "Context: The quick brown fox jumped over the lazy dog."
+            )
+        }
+    """
+    instruction = f"Human: {sample['instruction']}"
+    # By multiplying by a bool, we only add the context if it exists
+    context = f"Context: {sample['input']}" * bool(sample["input"])
+    return {
+        "formatted_input": f"{instruction} {context}",
+    }
+
+
+class S2SDatasetManager(BaseDatasetManager):
+    DEMO_DATASETS = [
+        "tatsu-lab/alpaca",
+        # "billsum",
+    ]
+
+    DATASET_FORMAT_FNS = {
+        "tatsu-lab/alpaca": format_alpaca,
+    }
+
+    DATASET_COLS = {
+        "tatsu-lab/alpaca": {
+            "input": "formatted_input",
+            "target": "output",
+        }
+    }
+
+    def _validate_dataset_dict(
+        self,
+        dd: DatasetDict,
+        inference_names: List[str],
+        labels: Optional[List[str]] = None,
+    ) -> DatasetDict:
+        """Validates the core components of the provided (or created) DatasetDict
+
+        The DatasetDict that the user provides or that we create from the provided
+        train/test/val data must have the following:
+            * all keys must be one of our valid key names
+            * it must have a `text` column
+            * it must have a `label` column
+                * if the `label` column isn't a ClassLabel, we convert it to one
+
+        We then also convert the keys of the DatasetDict to our `Split` key enum so
+        we can access it easier in the future
+        """
+        clean_dd = super()._validate_dataset_dict(dd, inference_names, labels)
+        for key in list(clean_dd.keys()):
+            ds = clean_dd.pop(key)
+            # TODO: temporary, update
+            ds = ds.select(range(100))
+
+            if "id" not in ds.features:
+                ds = ds.add_column("id", list(range(ds.num_rows)))
+            clean_dd[key] = ds
+        return add_val_data_if_missing(clean_dd, TaskType.seq2seq)
+
+
+def _get_labels(dd: DatasetDict, labels: Optional[List[str]] = None) -> List[str]:
+    """Gets the labels for this dataset from the dataset if not provided."""
+    if labels is not None and isinstance(labels, (list, np.ndarray)):
+        return list(labels)
+    train_labels = dd[Split.train].features["label"]
+    if hasattr(train_labels, "names"):
+        return train_labels.names
+    return sorted(set(dd[Split.train]["label"]))
+
+
+def _log_dataset_dict(
+    dd: DatasetDict, input_col: str = "text", target_col: str = "label"
+) -> None:
+    for key in dd:
+        ds = dd[key]
```

Comment:
I forget what HF datasets' structure is, but I'm almost sure you could do:

```suggestion
    for key, ds in dd.items():
```